### ISSUE TITLE
Cannot update ECS Agent

### ISSUE THREAD

**Issue by bendavies** [aws/amazon-ecs-agent#2287]
### Summary
Cannot update ECS agent to latest version.

### Description
We're currently running v1.32.1 on our cluster. ECS tells us a update is available: 
![image](https://user-images.githubusercontent.com/625392/69339432-3215c200-0c5d-11ea-928f-c0bfbf560a2a.png)
Trying to update to 1.33 via the UI and API both error:
![image](https://user-images.githubusercontent.com/625392/69339447-3d68ed80-0c5d-11ea-9e11-f84b0d4a48d7.png)

```
An error occurred (NoUpdateAvailableException) when calling the UpdateContainerAgent operation: There is no update available for your container agent.
```

Any ideas what could be wrong here?

Thanks!

**Comment by fenxiong** [comment]
Hi,
Are you using ECS Optimized AMI? If so what's the image id that you are using?

**Comment by fenxiong** [comment]
Not sure if it's your case, but one potential reason to get this is that you updated Docker on the instance to from 18.06 to 18.09 (latest available from Amazon Linux). In that case you will get this error.

**Comment by tedWexus** [comment]
I had a different problem, but the root of the problem was updating Docker to v18.09.9-ce in my EC2 instance. The following line in my auto scaling launch configuration User Data was responsible for the update:
```
yum update -y
```
I created a new launch configuration without that line in User Data, created the Auto Scaling Group, and the EC2 instance that was automatically created had Docker v18.06.1-ce installed. ECS Agent v1.33.0 was also installed! I am using ECS Optimized AMI.

**Comment by bendavies** [comment]
Indeed, we are using the ECS optimised AMI, but have updated docker with `yum update -y` too.

```
docker -v
Docker version 18.09.9-ce, build 039a7df
```

**Comment by bendavies** [comment]
We were able to rebuild our custom AMI from the ECS Optimized AMI which came with v1.33 of the ECS agent, and then update docker to 18.09.

The error is strange though. We've been on 18.09 for ages and have updated the agent via the UI without issue up until now.

**Comment by petderek** [comment]
@bendavies This issue should be resolved now, as of our latest AMI update. Please let us know if you are still seeing issues.

**Comment by petderek** [comment]
Ok, we've put out a fix and I'm no longer able to repro this condition. 😄 

We released agent 1.35.0, and the AMIs now ship with Docker 18.09 by default.
